### PR TITLE
Insert support in hive connector

### PR DESCRIPTION
**Pull request by stagraqubole**
Replacing previous PR (https://github.com/facebook/presto/pull/1433) with this. This has changes ported to new insert api

### CODE REVIEW

**Comment by nezihyigitbasi** [comment]
@stagraqubole @electrum any plans to finalize this PR?

**Comment by yuananf** [comment]
@stagraqubole  @nezihyigitbasi @electrum 
I have made insert work in hive connector based on this PR.
Will make another pull request recently.

**Comment by stagraqubole** [comment]
I am in middle of updating our presto code base and porting the Insert work to latest apis. I should be refreshing this PR around next week.

**Comment by stagraqubole** [comment]
To give some context, this particular feature has made Presto upgrades tricky for us at Qubole because of the large number number of conflicts. This has also prevented us from refreshing the PR recently. Hopefully, we can get this PR sorted out soon - this will benefit presto/hive users and it will help us rev faster as well.

**Comment by yuananf** [comment]
That's great! Hope for the new pull request!

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveClient.java [hunk 1]
```diff
@@ -1380,4 +1540,96 @@ public boolean apply(HivePartition partition)
...
+                    public void run()
+                    {
+                        rename(srcPath, destPath);
+                    }
+                });
+
+            }
+        }
+
+        executor.shutdown();
```
Comment (arhimondr) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveClient.java]: Probably you can use here either  invokeAll() or awaitTermination().

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveClient.java [hunk 2]
```diff
@@ -1380,4 +1540,96 @@ public boolean apply(HivePartition partition)
...
+        while (!executor.isTerminated()) {
+            try {
+                Thread.sleep(1000);
+            }
+            catch (InterruptedException e) {
+                // ignore
```
Comment (arhimondr) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveClient.java]: Do not swallow interrupted flag.

**Comment by stagraqubole** [comment]
Opened the new PR(#2574) addressing the review comments and porting changes to new interfaces.

**Pull request closed without merging**